invenio_deposit/views/rest.py

```diff
@@ -193,11 +194,8 @@ def post(self, pid, record, action):
         getattr(record, action)(pid=pid)
 
         db.session.commit()
-
-        # Index record.
-        if action == 'publish':
-            record.indexer.index(record.fetch_published()[1])
-
+        post_action.send(self, action=action, pid_value=pid.pid_value,
+                         deposit_uuid=record.id)
```

Comment:
The listener is fetching and indexing the published record.
The signal is sending the deposit information as input.